core/audits/layout-shifts.js

```diff
@@ -0,0 +1,150 @@
+  /** Label for a column in a data table; entries in this column will be a number representing how large the layout shift was. */
+  columnScore: 'Layout shift score',
+  /** A possible reason why that the layout shift occured. */
+  rootCauseUnsizedMedia: 'A layout shift may have occurred because of CSS defining the size of an otherwise unsized element',
+  /** A possible reason why that the layout shift occured. */
+  rootCauseFontChanges: 'A layout shift may have occurred because the font for some text changed',
+  /** A possible reason why that the layout shift occured. */
+  rootCauseInjectedIframe: 'A layout shift may have occurred because an iframe was added to the page without space being previously allocated for it',
+  /** A possible reason why that the layout shift occured. */
+  rootCauseRenderBlockingRequest: 'A layout shift may have occurred because of a render blocking request',
+};
+/* eslint-enable max-len */
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+class LayoutShifts extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'layout-shifts',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.METRIC_SAVINGS,
+      guidanceLevel: 2,
+      requiredArtifacts: ['traces', 'RootCauses', 'TraceElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const traceEngineResult = await TraceEngineResult.request({trace}, context);
+    const clusters = traceEngineResult.LayoutShifts.clusters ?? [];
+    const {cumulativeLayoutShift: clsSavings, impactByNodeId} =
+      await CumulativeLayoutShiftComputed.request(trace, context);
+
+    /** @type {Item[]} */
+    const items = [];
+    const layoutShiftEvents = clusters.flatMap(c => c.events);
+    for (const event of layoutShiftEvents) {
+      const biggestImpactNodeId = TraceElements.getBiggestImpactNodeForShiftEvent(
+        event.args.data.impacted_nodes, impactByNodeId);
+      const biggestImpactElement = artifacts.TraceElements.find(
+        t => t.traceEventType === 'layout-shift' && t.nodeId === biggestImpactNodeId);
+
+      // Turn root causes into sub-items.
+      const index = layoutShiftEvents.indexOf(event);
+      const rootCauses = artifacts.RootCauses.layoutShifts[index];
+      /** @type {SubItem[]} */
+      const subItems = [];
+      if (rootCauses) {
+        for (const cause of rootCauses.fontChanges) {
+          const url = cause.request.args.data.url;
+          subItems.push({
+            url,
+            cause: str_(UIStrings.rootCauseFontChanges),
+          });
+        }
+        for (const cause of rootCauses.iframes) {
```

Comment:
andres commented that "I remember this one getting noisy at some point on apps with many ads". Something for us to keep in mind.    Based on the impl, I don't think 0x0 iframes will be counted, but 1x1 are.